src/com/couchbase/versions/GithubVersions.groovy

```diff
@@ -101,6 +101,37 @@ class GithubVersions {
         throw new RuntimeException("Could not find the nearest tag for ${shaOrBranch} in ${repo}")
     }
 
+    static GithubSnapshotAttributes getSnapshotAttributesFromCommit(
```

Comment:
Could a better name for this be `getSnapshotAttributesUsingCommitAsReference` or something similar? The reason is to try to avoid anyone misusing this thinking that "commit" refers to the snapshot commit